### DISCUSSION TITLE
utilizzare dispositivi Zigbee

### DISCUSSION THREAD

**Discussion by PiuCurioso** [indomus/forum#558] [ZigBee, Z-Wave, Thread, Matter]
Carissimi buongiorno ho letto questo articolo https://indomus.it/progetti/adottare-ser2net-ed-esp-01-su-z-bee-duo-per-gestione-remota-con-zigbee2mqtt/ e volevo capire come metterlo in pratica per utilizzare le smart plug Osram in casa.

Però ho dei dubbi: se acquistare o meno un secondo Raspberry PI Zero W oppure no.e a questo proposito vi do qualche info sulla composizione della mia Lan: I dispositivi della Lan  sono  distribuiti su due edifici uno difronte all'altro in buona sostanza vi riassumo un po' come e' configurata la mia LAN: il modem router  che accede alla linea ADSL si trova in un edificio dove sono anche le ipcam e dove dovrei mettere le prese smart plug della Osram- da acquistare-(per accendere e spegnere le cam alla bisogna) il Raspberry PI zero W (che funge da server VPN )si trova in un altro edificio difronte a quello in cui c'è il router  (collegato all'ADSL).Nell'altro edificio ho portato la rete WiFi  grazie ad un ponte radio-tra i due edifici- Punto punto realizzato con due antenne Ubiquiti loco M2, Per cui sullo stesso  Raspberry che funge da server VPN(che si trova nell'edificio diverso da quello dove dovrei mettere le prese smart Osram Zigbee e in cui vi è il router con ADSL)   posso mettere zigbee2mqtt? mentre  ESP-01 su Z-Bee Duo lo posso mettere nell'altro edificio dove dovrei installare le prese Smart Zigbee(edificio dove è presente anche il router collegato all'ADSL), 
Per cui ricapitolabdo mi sto chiedendo se sia il Raspberry, su cui devo installare il gateway software zigbee2mqtt e sia ESP-01 su Z-Bee Duo devono stare fisicamente nello stesso appartamento ed essere collegati in Lan alla stessa wifi con lo stesso SSd etc in quanto attualmente il Raspberry PI zero W(server VPN) è collegato in wifi alla rete creata da un access point(con SSD diverso da quello della rete WiFi creata dal router nell'altro appartamento)collegato all'antenna Ubiquiti che funge da station (nell'edificio dove non c'è ilr collegato all'ADSL) con un certo SSD diverso dalla rete WiFi che sta nell'edificio dove c'è il router
.Non so se sono riuscito a spiegarmi😳Grazie comunque per le vostre delucidazioni Buona Domenica a tutti

**Comment by snakuzzo** [comment]
L'esp01 e zigbee2mqtt devono potrersi parlare via TCP...quindi devono stare sulla stessa LAN

**Reply by PiuCurioso** [reply]
Grazie mille Snakuzzo potresti gentilmente chiarirmi questa affermazione ?
Hai scritto "Se si vedono con rete diverse non credo, però se usi vpn basterebbe convogliare il tutto sotto la stessa in modo da sembrare tutta la stessa.
Cosa significa? E come si fa? Ad ogni modo i dispositivi si considerano della stessa Lan (anche se per essere collegati in Lan in modalità wifi, trovandosi in appartamenti diversi, usano access point diversi con un diverso SSD e una password diversa) se si "vedono" sono "pingabili" e quindi facenti parte tutti della stessa sottorete 255.255.255.0?(mia subnet)
Grazie ancora
 Buon pomeriggio

**Reply by snakuzzo** [reply]
Quando ti avrei scritto quello che dici ? Forse ti confondi. Se ti riferisci a Telegram, non sono io ad averti risposto.
In parole povere, dal server su cui è installato zigbee2mqtt devi poter raggiungere l'IP dell'ESP sulla porta impostata per la comunicazione.
Ad ogni modo... basta che si vedano...anche tramite VPN puoi farlo. Per far sì che l'ESP01 si colleghi in VPN ovviamente devi passare da un device che sia in grado di collegarsi ad una VPN. Solo con l'ESP01 non puoi farlo (credo).

`Zbee Duo con ESP01 -> router con client VPN -> internet -> server VPN -> LAN su cui è presente zigbee2mqtt e HA`

...Oppure dimentichi l'ESP01 e colleghi la chiavetta direttamente ad un RPI Zero W (come collegata [QUI](https://indomus.it/componenti/z-bee-duo-zigbee-coordinator-usb-hat-per-raspberry-pi-zero/))

`Zbee Duo su RPI Zero W con zigbee2mqtt + client VPN -> internet -> server VPN -> LAN su cui hai HA`

**Reply by PiuCurioso** [reply]
Grazie mille Snakuzzo scusami si mi riferivo a Telegram,👍 proverò la soluzione di questo articolo https://indomus.it/progetti/adottare-ser2net-ed-esp-01-su-z-bee-duo-per-gestione-remota-con-zigbee2mqtt/ dovrebbe funzionare in quanto la LAN è la stessa.Grazie ancora notte
